sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -112,19 +112,28 @@ void gnrc_ipv6_nib_init_iface(gnrc_netif_t *netif)
 #endif  /* GNRC_IPV6_NIB_CONF_SLAAC || GNRC_IPV6_NIB_CONF_6LN */
     _init_iface_router(netif);
 #if GNRC_IPV6_NIB_CONF_6LN
-    netif->flags |= GNRC_NETIF_FLAGS_6LN;
     netif->ipv6.rs_sent = 0;
 #endif  /* GNRC_IPV6_NIB_CONF_6LN */
-    if (netif->device_type == NETDEV_TYPE_IEEE802154) {
-        /* see https://tools.ietf.org/html/rfc6775#section-5.2 */
-        uint16_t src_len = IEEE802154_LONG_ADDRESS_LEN;
-        gnrc_netapi_opt_t opt = { .opt = NETOPT_SRC_LEN,
-                                  .data = &src_len,
-                                  .data_len = sizeof(src_len) };
-
-        /* XXX we are supposed to be in interface context here, so use driver
-         * directly everything else would deadlock anyway */
-        netif->ops->set(netif, &opt);
+    switch (netif->device_type) {
+        case NETDEV_TYPE_IEEE802154:
+            /* see https://tools.ietf.org/html/rfc6775#section-5.2 */
+            uint16_t src_len = IEEE802154_LONG_ADDRESS_LEN;
```

Comment:
Compilation error since a variable cannot be declared inside a switch.

Reply:
Arghs sorry ... should have at least compile tested -.-

Reply:
Fixed by adding braces around.